rclpy/src/rclpy/utils.cpp

```diff
@@ -89,4 +97,115 @@ convert_to_py(void * message, py::object pyclass)
   return py::reinterpret_steal<py::object>(convert(message));
 }
 
+const char *
+get_rmw_implementation_identifier()
+{
+  return rmw_get_implementation_identifier();
+}
+
+void
+assert_liveliness(py::capsule pyentity)
+{
+  if (0 == strcmp("rclpy_publisher_t", pyentity.name())) {
+    auto publisher = static_cast<rclpy_publisher_t *>(
+      rclpy_handle_get_pointer_from_capsule(
+        pyentity.ptr(), "rclpy_publisher_t"));
+    if (nullptr == publisher) {
+      throw py::error_already_set();
+    }
+    if (RCL_RET_OK != rcl_publisher_assert_liveliness(&publisher->publisher)) {
+      throw RCLError("Failed to assert liveliness on the Publisher");
+    }
+  } else {
+    throw py::type_error("Passed capsule is not a valid Publisher.");
+  }
+}
+
+py::list
+remove_ros_args(py::object pycli_args)
+{
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+
+  std::vector<const char *> arg_values;
+  const char ** const_arg_values = NULL;
+  py::list pyargs;
+  if (!pycli_args.is_none()) {
+    pyargs = pycli_args;
+    arg_values.resize(pyargs.size());
+    for (size_t i = 0; i < pyargs.size(); ++i) {
+      // CPython owns const char * memory - no need to free it
+      arg_values[i] = PyUnicode_AsUTF8(pyargs[i].ptr());
+      if (!arg_values[i]) {
+        throw py::error_already_set();
+      }
+    }
+    const_arg_values = &(arg_values[0]);
```

Comment:
@gbalke we can't index `arg_values` if it happens to be empty ! 

This is what slipped through the cracks in previous PRs and is causing issue in Windows Debug. Will send a patch fixing other occurrences, mind to fix this one as well?

Reply:
How do we want to handle this?
```cpp
if (!pycli_args.is_none() && 0 != pyargs.size())
```
or
```cpp
if (0 == pyargs.size()) {
  throw py::value_error("No arguments given")
}
```
?

Reply:
Or are we handling the case where resize fails?

Reply:
See https://github.com/ros2/rclpy/pull/740. It's fine for no arguments to be given. If resize fails, it'll throw.